text/287-cli-deprecation-warnings.md

```diff
@@ -0,0 +1,190 @@
+---
+rfc pr: [#290](https://github.com/aws/aws-cdk-rfcs/pull/290)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/287
+---
+
+# CDK CLI deprecation warnings
+
+If an element in the CDK Construct Library (class, interface, property, function) is deprecated,
+using it inside a CDK application should produce a warning in the CDK CLI for all commands that perform synthesis
+(`cdk synth`, `cdk diff`, `cdk deploy`, etc.).
+
+This is done in order to aid the migration from `V1` of the CDK to `V2`
+(where all deprecated elements from `V1` will be removed).
+
+## Working Backwards
```

Comment:
Can you add a working backwards section from the "contribution guide" which explains the contributor's experience (I.e you add @deprecated, and then add a decorator etc)?

Reply:
Done!